### PR TITLE
Keyboard Event Which is Depreciated 

### PR DESCRIPTION
**Pull request by GeorgeWL**
https://caniuse.com/keyboardevent-which

Though it's usable, it's been depreciated for a while, so is bad practice to use.

### CODE REVIEW

File: src/helpers/keycodes.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-    UP: '38',
+    DOWN: 'Down',
+    END: 'End',
+    ENTER: 'Enter',
+    HOME: 'Home',
+    LEFT: 'ArrowLeft',
+    RIGHT: 'ArrowRight',
+    SPACE: 'Space',
+    UP: 'ArrowUp',
+    ESCAPE:'Escape'||'Esc' // ie11 uses Esc instead of Escape.
```
Comment (joris-rotteveel) [inline src/helpers/keycodes.ts:10]: Thank you for this PR GeorgeWL!

I'd imagine that the `ESCAPE` key will always have the value `Escape`, so it won't work on IE11. This is a quick browser test (not in TS):
image

Maybe we can add a `ESCAPE_IE11` key in the map?

**Review by joris-rotteveel — CHANGES_REQUESTED** [review]
Not sure `ESCAPE` will ever hold the value `Esc`

File: src/helpers/keycodes.ts (hunk 1 above)
Comment (GeorgeWL) [reply src/helpers/keycodes.ts:10]: ah yeah you're right, i forgot it doesn't compare boolean strings like that
 I'll make that change

**Comment by GeorgeWL** [comment]
> Not sure `ESCAPE` will ever hold the value `Esc`

ie11 specifically does. It's a piece of non-compat in the Internet Explorer spec.

File: src/helpers/keycodes.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+    DOWN: 'Down',
+    END: 'End',
+    ENTER: 'Enter',
+    HOME: 'Home',
+    LEFT: 'ArrowLeft',
+    RIGHT: 'ArrowRight',
+    SPACE: 'Space',
+    UP: 'ArrowUp',
+    ESCAPE:'Escape',
+    ESCAPE_IE11:'Esc' // ie11 uses Esc instead of Escape.
```
[the remark this answers is not in the document]
Comment (joris-rotteveel) [reply src/helpers/keycodes.ts:11]: Ha, good catch Matthew. @GeorgeWL, for clarity, could you please remove those 2? I didn't spot we are not using the Escape keycode in our code at all.

**Review by joris-rotteveel — CHANGES_REQUESTED** [review]

File: src/helpers/keycodes.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (GeorgeWL) [reply src/helpers/keycodes.ts:11]: > Why was Escape added? 
> 
> It's good to have this as a constant but where is this constant used?

You know, I don't know why I added it, think it's cause I was using it in something else I'd made recently and was running on automatic 🤣

[the remark this answers is not in the document]
Comment (holloway) [reply src/helpers/keycodes.ts:11]: @GeorgeWL happy to merge this PR once this bit is deleted :)

**Review by joris-rotteveel — APPROVED** [review]

**Comment by joris-rotteveel** [comment]
@GeorgeWL Thanks for your contribution!

**Comment by joris-rotteveel** [comment]
@GeorgeWL have merged but noticed our tests were failing. The correct value for the SPACE key should be either ` ` or `spacebar`, check https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values for details

**Comment by GeorgeWL** [comment]
ah darn it, i can make a new pr to fix if you like, unless you already have fixed that?

**Comment by joris-rotteveel** [comment]
I have fixed it on our end, but because you mentioned you copied it from another of your projects I thought you might wanted to know. Thanks again for your work!

**Pull request merged**